### PR TITLE
Add null checks to min, max, and count UDAFs

### LINKED ISSUE #2244: NullPointerException occurs for min()/max() for Long or Double column containing null value
A `NullPointerException` occurs when `min()` and `max()` are called on Double or Long columns containing null values -- I would expect `min()` and `max()` to ignore null values. Also, `count(<column>)` returns the total number of rows _including_ the rows where `<column>=null` -- I would expect `count(<column>)` to return the number of non-null rows (`count(*)` should return the total: non-null and null).

To reproduce the NullPointerException:

First, create a topic 'output-topic' with a message with a non-null `bid`, with an associated stream:
```
docker run --interactive --rm --network tutorials_default confluentinc/cp-kafkacat kafkacat -b kafka:39092 -t output-topic -K: -C -o 0 
ticker_1:{"exchange":10, "bid":1.0, "time":1}

ksql> CREATE STREAM output_stream \
> (exchange int, bid double, time bigInt) \
>  WITH (kafka_topic='output-topic', value_format='JSON');
```
When performing an aggregate to get min(bid), 1.0 is returned, as expected:
```
ksql> select rowkey, min(bid) from output_stream group by rowkey;
ticker_1 | 1.0
```
However, if I produce a message with a null bid:
```
$ docker run --interactive --rm --network tutorials_default confluentinc/cp-kafkacat kafkacat -b kafka:39092 -t output-topic -K: -C -o 0 
ticker_1:{"exchange":10, "bid":1.0, "time":1}
ticker_1:{"exchange":10, "bid":null, "time":1}
```
The select query now results in a NullPointerException (the first line with 1.0 was from the query before producing a message with null):
```
ksql> select rowkey, min(bid) from output_stream group by rowkey;
ticker_1 | 1.0
java.lang.NullPointerException
Query terminated
```
I believe null checks should be added for `udaf/min/DoubleMinKudaf.java`, `udaf/max/DoubleMaxKudaf.java`, `udaf/min/LongMinKudaf.java`, `udaf/max/LongMaxKudaf.java`, and `udaf/count/CountKudaf.java`. 

I will submit a PR for the above null checks.

### PR DESCRIPTION
**Pull request by amleung21**
### Description 
Fixes #2244 by adding the appropriate checks to ignore `null` values.
- add `null` checks to min and max UDAFs for `Double` and `Long` and count UDAF so `null` values do not impact those aggregations.
- Change min and max UDAFs for `Double` and `Long` to use `Math.min()` and `Math.max()` to be consistent with `Integer` UDAFs.
- add documentation for aggregate function handling of `null` values.

### Testing done 
Unit tests added to test `null` values for each combination where checks were added. 

Manual testing building ksql-server and client:
- tested original `min(<column>)`, `max(<column>)`,  and `count(<column>) for `Double` with `null` columns and verified no longer encountering `NullPointerException` and that correct value is returned.
- verify that `count(*)` returns count of all rows, including rows with `null` value(s).

### Reviewer checklist
- [x] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [x] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-engine/src/main/java/io/confluent/ksql/function/udaf/count/CountKudaf.java [hunk 1]
```diff
@@ -37,6 +37,9 @@
 
   @Override
   public Long aggregate(final Object currentValue, final Long aggregateValue) {
+    if (currentValue == null) {
+      return aggregateValue;
+    }
```
Comment (hjafarpour) [inline ksql-engine/src/main/java/io/confluent/ksql/function/udaf/count/CountKudaf.java]: We should count nulls in `count` function. If we need to count non- null values only we can easily add a filter in `WHERE` clause.

File: ksql-engine/src/main/java/io/confluent/ksql/function/udaf/count/CountKudaf.java [hunk 2]
```diff
@@ -47,6 +50,9 @@ public Long aggregate(final Object currentValue, final Long aggregateValue) {
 
   @Override
   public Long undo(final Object valueToUndo, final Long aggregateValue) {
+    if (valueToUndo == null) {
+      return aggregateValue;
+    }
```
Comment (hjafarpour) [inline ksql-engine/src/main/java/io/confluent/ksql/function/udaf/count/CountKudaf.java]: Same here. we should count the nulls.

File: ksql-engine/src/main/java/io/confluent/ksql/function/udaf/count/CountKudaf.java (hunk 1 above)
Comment (amleung21) [reply ksql-engine/src/main/java/io/confluent/ksql/function/udaf/count/CountKudaf.java]: @hjafarpour My thinking was to emulate what most databases do where `count(*)` returns the count regardless of nulls and `count(<column>)` returns the count where `<column>` is not null.

Comment (amleung21) [reply ksql-engine/src/main/java/io/confluent/ksql/function/udaf/count/CountKudaf.java]: @hjafarpour Do you agree that being consistent with the behavior of most RDBMS's (e.g. Oracle, MySQL, Postgres, SQL Server, DB2), by ignoring `null`, is desirable in this case?

Comment (dguy) [reply ksql-engine/src/main/java/io/confluent/ksql/function/udaf/count/CountKudaf.java]: I'm in agreement that we should be consistent with the DB world as much as possible

**Comment by dguy** [comment]
@amleung21 can you rebase against master again? I would click update branch but it doesn't always work out too well! Thanks

**Comment by amleung21** [comment]
@dguy rebased! Thanks.

**Comment by apurvam** [comment]
Thanks for your first contribution @amleung21. I just merged your commit :)

**Pull request merged**